src/main/java/pl/bajerska/befindyourmeal/edamam/ApiServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package pl.bajerska.befindyourmeal.edamam;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import pl.bajerska.befindyourmeal.recipe.RecipeCriteria;
+
+@Service
+public class ApiServiceImpl implements ApiService {
+
+
+    @Override
+    public String test() {
+        RestTemplate rest = new RestTemplate();
+
+        ResponseEntity<String> exchange = rest.exchange(
+                "https://api.edamam.com/search?app_id=2f71b0fa&app_key=c0ed4fc9f52cd3c0d3d1f74b8f6e220d&q=haddock",
+                HttpMethod.GET,
+                HttpEntity.EMPTY,
+                String.class);
+        return exchange.getBody();
+    }
+
+    private String queryBuilder(RecipeCriteria recipeCriteria) {
+        String query = new String();
+        for (String ing : recipeCriteria.getIngredients()) {
+            if ((query.length() > 0) && (ing.length() > 0))
+                query += ",";
+            query += ing;
```

Comment:
Shouldn't it be a block of code?
Please consider StringBuilder (you can check the differences between StringBuilder and StringBuffer, it's quite common question on job interviews). In the next step we can work towards lambda expression here, but I feel it should be next step. :)

Reply:
I made lambdas. Please take a look.

Reply:
Cool! I think there is small issue now, the string will have comma as the end character. But it can be fixed. :) https://www.baeldung.com/java-string-joiner -> point 5 :)